Documentation/driver-api/ancillary_bus.rst

```diff
@@ -0,0 +1,218 @@
+.. SPDX-License-Identifier: GPL-2.0-only
+
+=============
+Ancillary Bus
+=============
+
+See <linux/ancillary_bus.h> for the models for ancillary_device and
+ancillary_driver.
+
+In some subsystems, the functionality of the core device (PCI/ACPI/other) may
+be too complex for a single device to be managed as a monolithic block or
+a part of the functionality might need to be exposed to a different subsystem.
+Splitting the functionality into smaller orthogonal devices would make it
+easier to manage data, power management and domain-specific interaction with
+the hardware. A key requirement for such a split is that there is no dependency
+on a physical bus, device, register accesses or regmap support. These
+individual devices split from the core cannot live on the platform bus as they
+are not physical devices that are controlled by DT/ACPI. The same argument
+applies for not using MFD in this scenario as MFD relies on individual function
+devices being physical devices that are DT enumerated.
+
+An example for this kind of requirement is the audio subsystem where a single
+IP may handle multiple entities such as HDMI, Soundwire, local devices such as
+mics/speakers etc. The split for the core's functionality can be arbitrary or
+be defined by the DSP firmware topology and include hooks for test/debug. This
+allows for the audio core device to be minimal and focused on hardware-specific
+control and communication.
+
+The ancillary bus is intended to be minimal, generic and avoid domain-specific
+assumptions. Each ancillary_device represents a part of its parent
+functionality. The generic behavior can be extended and specialized as needed
+by encapsulating an ancillary_device within other domain-specific structures and
+the use of .ops callbacks. Devices on the ancillary bus do not share any
+structures and the use of a communication channel with the parent is
+domain-specific.
+
+When Should the Ancillary Bus Be Used
+=====================================
+
+One example could be a multi-port PCI network interface card whose driver will
+allocate and register an ancillary_device for each physical function on the NIC.
+An RDMA driver, in another subsystem of the kernel, registers an
+ancillary_driver that will be matched with and probed for each of these
+ancillary_devices.  This will give the RDMA driver acces to the shared data/ops
+in the PCI drivers shared object.
+
+Another usage case is for a PCI device whose functionality is complex enough
```

Comment:
@dmertman @shirazsaleem 

Please review above three use cases, I think you can omit lengthy description below about when to use it.
3 examples are more than enough. At least (2) and (3) are discussed many times so no need to in detail to say about ancillary_device and ancillary_driver object. Its inherent to linux kenel device model.